blockchain/backup/NFTtheWorld.sol

```diff
@@ -13,46 +15,66 @@ contract NFTtheWorld {
     uint256[] public dropHashes;
 
     struct NFTOwnership {
-        address owner;
-        uint256 nftId;
+        address payable owner;
+        string uri;
         uint256 dropId;
         uint256 dropTime;
         address reservedFor;
+        uint256 weiPrice;
     }
 
+    // For testing reasons this is public
+    // To check if an address is an admin
+    mapping(address => bool) public adminRights;
+    // Dictionary of form <dropHash>: list of NFTOwnerships
     mapping(uint256 => NFTOwnership[]) public nftOwnerships;
-    //Dictionary of form <user address>: <<dropHash>: <number of reserved NFTs>>
+    // Dictionary of form <user address>: <<dropHash>: <number of reserved NFTs>>
     mapping(address => mapping(uint256 => uint256)) public nftReservations;
     // Dictionary of form <dropHash>: <list of nftHashes>
-    mapping(uint256 => uint256[]) public availableNFTs;
+    mapping(uint256 => string[]) public availableNFTs;
     // Dictionary of form  <dropHash>: <nftCount>
     mapping(uint256 => uint256) public availableNFTsCount;
     // Dictionary of form <dropHash>: <maximal number of NFTs a user can reserve/buy from this drop>
-    mapping(uint256 => uint256) public maxNumberOfNFTsToBuy;
+    mapping(uint256 => uint256) private maxNumberOfNFTsToBuy;
     // Dictionary of form <dropHash>: <number of NFTs that have been requested by users>
     mapping(uint256 => uint256) public reservedNFTsCount;
 
-    mapping(address => uint256[]) public nftReservationInformationOfUsers;
+    mapping(address => mapping(uint256 => string[]))
+        private nftReservationInformationOfUsers;
     mapping(address => uint256[]) public nftAssetsInformationOfUsers;
 
     // Used to track which addresses have joined the drop
-    mapping(uint256 => address[]) public joinedUsers;
+    mapping(uint256 => address[]) private joinedUsers;
+
+    //TODO add all of us in list of admins
+    constructor() {
+        adminRights[msg.sender] = true;
+    }
 
     // This function lets a user create a drop by specifiyng a drop time and the number of available NFTs.
     // During the creation of the drop, the maximum number of NFTs a user can reserve/buy in this drop is calculated.
     // It is one for a total number of NFTs lower than 20 and 5% otherwise.
-    function createDrop(uint256 _dropTime, uint256 _numberOfNFTS) public {
+    function createDrop(
+        uint256 _dropTime,
+        uint256 _numberOfNFTS,
+        string[] memory _uris,
+        uint256 _weiPrice
+    ) public onlyByAdmins {
         uint256 dropHash = generateRandomNumber(_dropTime);
```

Comment:
Would it not be easier to just count the number of drops happened? Then, accessing the different drops also becomes easier and maybe we could use an array instead of a mapping for the drops or just remember the total number of created drops?

Reply:
good idea, will change